include/aws/common/error.h

```diff
@@ -0,0 +1,104 @@
+#ifndef AWS_COMMON_ERROR_H_
+
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#define AWS_COMMON_ERROR_H_
+
+#include <aws/common/exports.h>
+#include <stdint.h>
+
+struct aws_error_info {
+    int error_code;
+    const char *literal_name;
+    const char *error_str;
+    const char *lib_name;
+    const char *formatted_name;
+};
+
+struct aws_error_info_list {
+    const struct aws_error_info *error_list;
+    uint16_t count;
+};
+
+#define AWS_DEFINE_ERROR_INFO(NAME, C, ES, LN)                                                                         \
+{                                                                                                                      \
+    .literal_name = #C,                                                                                                \
+    .error_code = C,                                                                                                   \
+    .error_str = ES,                                                                                                   \
+    .lib_name = LN,                                                                                                    \
+    .formatted_name = LN ": " #C ", " ES                                                                               \
+}
+
+typedef void(*aws_error_handler)(int err, void *ctx);
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * Returns the latest error code on the current thread, or 0 if none have occurred.
```

Comment:
Does this function clear the error code, or does it leave it there?

Reply:
does not clear it, just retrieves it. aws_reset_error() resets it